### DISCUSSION TITLE
Authentication with Amazon ALB

### DISCUSSION THREAD

**Discussion by ycastorium** [ravendb/ravendb#14380] [Q&A] [answered]
Good morning, i want to use Amazon ALB to be a load balancer of my cluster, sadly Amazon Application Load Balancer does not support mTLS, that means the TLS/SSL Handshake ends at the LB itself, never propagating the ssl downstream, making it impossible to run use the client authentication as is. 

I have some options: 

- Swapping ALB to a NLB which propagates the SSL down, but it's not as nice as ALB. 
- Let RavenDB run unauthenticated under the LB and control the access directly on the load balancer (not as fine grained) 
- Have a NGINX in-between with a self-signed cert, make raven trust this cert and send the client-cert in a header and unwrap it on Nginx (Sounds icky) 

So i have a question, can RavenDB read the client-cert from a header? So i can bypass the nginx mumbo jumbo?

**Comment by ayende** [comment, marked as answer]
That is very much by design. 

1. Never run without authentication, see: https://ravendb.net/articles/defense-in-depth-and-the-local-server
2. Why are you running load balancer in the first place. You don't need it and it complicates things. RavenDB uses different endpoints for different nodes, and load balancing, failover, etc are all handled for you. Adding a load balancer in the middle complicate things, violate assumptions we make, etc.
3. Adding a trusted middle-man means that you lose features such as the ability to limit clients to particular databases, auditing, etc. 

Before you go in that route, start from this: Why are you even going with load balancer here?

**Reply by ycastorium** [reply]
Thank you! I agree, my initial idea was to have my RavenDB cluster under Fargate pop-up the cluster-nodes as tasks and let the ALB deal with routing, since that's a bad idea i'll just have a Nginx or HAProxy between my cluster and set it up in a similar way to a Kubernetes deploy. 

Thanks!
